public/js/app.Ctrl.js

```diff
@@ -0,0 +1,201 @@
+                    break;
+                 // correct input and round is completed
+                 case 2:
+                    round++;
+                    resetRound(true);
+                    break;
+                  // correct input and sequence is completed
+                  case 3:
+                    win.play();
+                    toggleLights(0, 5);
+                    break;
+             }
+         }
+    };
+
+    $scope.lightUp = function(n) {
+        if(on === true && lock === false) {
+            $scope[colorCode[n]] = AppService.getLightColorObj(n);
+            playSound(n);
+        }
+    };
+
+    turnOnAllLights = function() {
+        $scope[colorCode[1]] = AppService.getLightColorObj(1);
+        $scope[colorCode[2]] = AppService.getLightColorObj(2);
+        $scope[colorCode[3]] = AppService.getLightColorObj(3);
+        $scope[colorCode[4]] = AppService.getLightColorObj(4);
+    };
+
+    turnOffAllLights = function() {
+        $scope[colorCode[1]] = AppService.getDarkColorObj(1);
+        $scope[colorCode[2]] = AppService.getDarkColorObj(2);
+        $scope[colorCode[3]] = AppService.getDarkColorObj(3);
+        $scope[colorCode[4]] = AppService.getDarkColorObj(4);
+    };
+
+    // resets everything and generates a new sequence
+    reset = function(restart) {
+        AppService.resetSequence();
+        AppService.generateSequence(AppService.getSequenceLength());
+        // initialize sequence of color objects
+        colorSequence = AppService.getColorSequence();
+        resetToRound1(restart);
+    };
+
+    // resets back to round 1
+    resetToRound1 = function(restart) {
+        cancelAllTimers();
+        turnOffAllLights();
+        $scope.screen = "Go!";
+        round = 1;
+        resetRound(restart);
+    };
+
+    //resets the current round
+    resetRound = function(restart) {
+        count = 0;
+        AppService.resetUserInput();
+        if(restart) {
+            var timer = $timeout(function() {
+                startRound(round);
+            }, speed*3);
+        }
+    };
+
+    wrongInput = function() {
+        buzzer.play();
+        $scope.screen = '!!';
+        lock = true;
+        toggleLights(0, 3);
+        lock = false;
+        var timer = $timeout(function() {
+            if(strict === false)
+                resetRound(true);
+            else resetToRound1(true);
+        }, 1500);
+    };
+
+    toggleLights = function(i, n) {
+        if(i < n) {
+            turnOnAllLights();
+            var timer1 = $timeout(function() {
+                turnOffAllLights();
+                var timer2 = $timeout(function() {
+                    toggleLights(i+1, n);
+                }, 200);
+            }, 200);
+        }
+    };
+
+    cancelAllTimers = function() {
+        $timeout.cancel(turnOffTimer);
+        $timeout.cancel(recursiveTimer);
+        $timeout.cancel(timer);
+    };
+
+    playSound = function(n) {
+        sound[n].play();
+    };
+
```

Comment:
Use a loop instead of repeating code.  Even then, this wouldn't scale.  Even if your app audio grows only linearly (you start adding different music packages that user selects), you would have all this extra code in your app controller managing all this.  Even worse, these links are hard coded.  Once you have more than 10 links, it starts becoming unmanageable.  Create an angular config and include a config section for audio to store the link names.  Add an audio service '/js/services/audio.service.js' with some methods to get stuff.  Like getWinAudioElement(), which returns an element with the src already set.